### ISSUE TITLE
Cross compiling a BE kernel on a LE distro fails in arch/powerpc/boot

### ISSUE THREAD

**Issue by antonblanchard** [linuxppc/linux#29]
Looks like we aren't passing -mbig-endian to gcc when building the 32bit VDSO and we end up with little endian object files:

```
/usr/bin/ld: arch/powerpc/kernel/vdso32/sigtramp.o: compiled for a little endian system and target is big endian
```

**Comment by mpe** [comment]
Where did you hit this?

I can cross build big endian fine on Ubuntu 15.10, using their cross toolchain:

```
ii  gcc-powerpc-linux-gnu           4:5.2.1-1            ppc64el              The GNU C compiler for powerpc architecture
```

**Comment by npiggin** [comment]
Should be fixed in mpe's tree

https://git.kernel.org/powerpc/c/164af597ce945751e2dcd53d0a

**Comment by mpe** [comment]
The VDSO is fixed in 164af597ce945751e2dcd53d0a86e84203a6d117.

But we then fail in `arch/powerpc/boot`.

**Comment by npiggin** [comment]
Looks like drivers/crypto/vmx/aesp8-ppc.S also fails to be updated on endian switch. Also probably needs a FORCE/if_changed

**Comment by npiggin** [comment]
How about now?

**Comment by mpe** [comment]
`arch/powerpc/boot` fixed in f0f7fe1ac353.

Final breakage is in `drivers/crypto/vmx`, patch for that is being tested now.

**Comment by deece** [comment]
Built ppc64_defconfig & pseries_defconfig on ppc64le successfully.

@mpe Please close.
